indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/supervisor/SeekableStreamSupervisor.java

```diff
@@ -405,9 +394,16 @@ public String getType()
     }
 
     @Override
-    public boolean canAlsoHandle(Notice otherNotice)
+    public int hashCode()
     {
-      return otherNotice.getType().equals(TYPE);
+      return 0;
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+      // All RunNotices are the same. They are de-duplicated on insertion into the NoticesQueue "notices".
+      return obj != null && obj.getClass().equals(RunNotice.class);
```

Comment:
Makes sense. 

Since the RunNotice has no state, I wonder if we shouldn't just have a singleton like so:
`static final RunNotice RUN_NOTICE = new RunNotice()` and just use that wherever needed `addNotice(RUN_NOTICE)`.

It would also cleanup the time-based condition in `handle`, in that the `lastRunTime` can now be a field in `RunNotice` rather than in the `Supervisor`. Maybe this is not even needed now since we are already doing aggressive deduplication in the queue itself.

Reply:
I think a singleton makes sense, I can add that tomorrow when fixing the build issues.

However then `lastRunTime` couldn't be a field in `RunNotice`, since if `RunNotice` is a singleton, it's a singleton across all supervisors for all datasources. I'll look into whether it's still needed at all.

Reply:
Oh, nevermind, it can't be `static` singleton since it accesses state of the outer class. I'll just leave it the way it is. I'm not sure if we still need the `lastRunTime`, so I'll leave that too.

Reply:
Ah, of course, we still need to call the actual `runInternal` 😅